// ToiChat Buffer Protocol 
// Messages
//
// Created on: 02/01/2016
// Author: Toi-Group
//
package ToiChatProtocol;

message ToiChatMessage {
    oneof messageType {
        DnsMessage dnsMessage = 1;
        ServerMessage serverMessage = 2;
        OneToOne oneToOneMessage = 3;
    }
}

message DnsMessage {
    // Gather message about client sending a new message
    //
    required string clientName = 1; // Hostname/CallSign
    required int32 clientId = 2; // IPaddress
    required int32 lastUpdate = 3;
    optional string description = 4 [default = "NONE"]; // Misc Info.
}

message ServerMessage {
    required string operation = 1;
    optional string description = 2 [default = ""];
}

message OneToOne {
    required string id = 1;
}